### PR TITLE
Provide swaption pricing across cash and physical

### PR DESCRIPTION
**Pull request by jodastephen**
Single entry point for product and trade.

Javadoc in new code copied from existing pricers.
See #847

### CODE REVIEW

File: modules/pricer/src/main/java/com/opengamma/strata/pricer/swaption/VolatilitySwaptionTradePricer.java [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+   * <p>
+   * The present value sensitivity of the product is the sensitivity of the present value to
+   * the underlying curves.
+   * 
+   * @param trade  the swaption trade
+   * @param ratesProvider  the rates provider
+   * @param swaptionVolatilities  the volatilities
+   * @return the present value curve sensitivity of the swap trade
+   */
+  public PointSensitivityBuilder presentValueSensitivityStickyStrike(
```
Comment (traggatt) [inline modules/pricer/src/main/java/com/opengamma/strata/pricer/swaption/VolatilitySwaptionTradePricer.java]: Is it documented anywhere what the expected unit for a call like this is? As a user I would find it useful for the method to state what the magnitude of the sensitivity is, e.g. to 1bp, to 1%, 100% etc. If not stated, I would probably assume 1bp, though I know this isn't always the case at the pricer level...

Comment (traggatt) [inline modules/pricer/src/main/java/com/opengamma/strata/pricer/swaption/VolatilitySwaptionTradePricer.java]: Might also be useful, in this case, to state what the implication of the sticky strike is.

File: modules/pricer/src/main/java/com/opengamma/strata/pricer/swaption/VolatilitySwaptionTradePricer.java [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+  public MultiCurrencyAmount currencyExposure(
+      ResolvedSwaptionTrade trade,
+      RatesProvider ratesProvider,
+      SwaptionVolatilities swaptionVolatilities) {
+
+    return MultiCurrencyAmount.of(presentValue(trade, ratesProvider, swaptionVolatilities));
+  }
+
+  /**
+   * Calculates the current of the swaption trade.
```
Comment (traggatt) [inline modules/pricer/src/main/java/com/opengamma/strata/pricer/swaption/VolatilitySwaptionTradePricer.java]: current cash

File: modules/pricer/src/main/java/com/opengamma/strata/pricer/swaption/VolatilitySwaptionTradePricer.java [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+    ResolvedSwaption product = trade.getProduct();
+    CurrencyAmount pvProduct = productPricer.presentValue(product, ratesProvider, swaptionVolatilities);
+    Payment premium = trade.getPremium();
+    CurrencyAmount pvPremium = paymentPricer.presentValue(premium, ratesProvider);
+    return pvProduct.plus(pvPremium);
+  }
+
+  //-------------------------------------------------------------------------
+  /**
+   * Computes the currency exposure of the swaption trade
```
Comment (traggatt) [inline modules/pricer/src/main/java/com/opengamma/strata/pricer/swaption/VolatilitySwaptionTradePricer.java]: Missing full stop! 😳

File: modules/pricer/src/main/java/com/opengamma/strata/pricer/swaption/VolatilitySwaptionTradePricer.java (hunk 1 above)
Comment (jodastephen) [inline modules/pricer/src/main/java/com/opengamma/strata/pricer/swaption/VolatilitySwaptionTradePricer.java]: Will deal with units in a later PR.

**Pull request merged**